### PR TITLE
S3C-3526 release documentation

### PR DESCRIPTION
**Pull request by tcarmet**

### CODE REVIEW

File: docs/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+only for the CI. The CI will push images with every CI build tagging the
+content with the developer’s branch short SHA-1 commit hash. This allows those
+images to be used by developers, CI builds, build chain and so on.
+
+## How to pull docker images
+
+```sh
+    docker pull registry.scality.com/vmware-ose-scality-dev/vmware-ose-scality:<short SHA-1 commit hash>
+    docker pull registry.scality.com/vmware-ose-scality/vmware-ose-scality:<tag>
+```
```
Comment (miniscruff) [inline docs/RELEASE.md:22]: I might of wanted to mention this before now since you have made a few of these already... but unless we can get Harbot to show us the list of public tags ( eg https://registry.scality.com/harbor/projects/25/repositories/cloudserver%2Fcloudserver ) without authenticating it becomes a bit hard to use. Sure <tag> and <commit hash> are descriptive but unless you can see the actual list you are sort of having to play magic string bingo that the git tag just so happens to line up with our docker image. 
And yes, this should always line up but it doesn't mean it always will, a broken build or deleted tag on one side could always happen.

Obviously dev images are whatever as it is use at your own risk sort of thing which is fine ( although a 'latest-dev' would be nice to avoid looking it up )

Basically, we need a page like this https://hub.docker.com/_/mongo?tab=tags per public project.

Comment (tcarmet) [reply docs/RELEASE.md:22]: good question, indeed those are a bit painful, :thinking:

We're a bit outdated on harbor's version and may be missing (or not) some features towards that, I can have a look.

File: docs/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Vault Release Plan
+
+## Docker Image generation
+
+Docker images are hosted on [registry.scality.com](registry.scality.com).
```
[the remark this answers is not in the document]
Comment (tcarmet) [reply docs/RELEASE.md:5]: true!
```suggestion
Docker images are hosted on registry.scality.com.
```

File: docs/RELEASE.md (hunk 1 above)
Comment (miniscruff) [reply docs/RELEASE.md:22]: You are able to search without logging in, but when you click a project it forces you to sign in. So, it is like half there...

**Review by saitharunthodupunuru — APPROVED** [review]

File: docs/RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Vault Release Plan
```
Comment (saitharunthodupunuru) [inline docs/RELEASE.md:1]:
```suggestion
# Vmware-ose-scality Release Plan
```

File: docs/RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Docker Image generation
+
+Docker images are hosted on registry.scality.com.
+It has two names spaces for vault:
```
Comment (saitharunthodupunuru) [inline docs/RELEASE.md:6]:
```suggestion
It has two names spaces for vmware-ose-scality:
```

**Review by saitharunthodupunuru — CHANGES_REQUESTED** [review]

File: docs/RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+## Release Process
+
+To release a production image:
+
+* Decide an appropriate tag which will be used to tag the repository and
+the docker image.
+* Update the version in `build.gradle` file with the same release tag.
+* Create a PR and merge the `build.gradle` change.
+* Tag the repository using the same tag.
+* With the below parameters, [force a build](https://eve.devsca.com/github/scality/vault/#/builders/bootstrap/force/force):
```
Comment (saitharunthodupunuru) [inline docs/RELEASE.md:33]:
```suggestion
* With the below parameters, [force a build](https://eve.devsca.com/github/scality/vmware-ose-scality/#/builders/bootstrap/force/force):
```

**Review by saitharunthodupunuru — CHANGES_REQUESTED** [review]

File: docs/RELEASE.md (hunk 1 above)
Comment (tcarmet) [reply docs/RELEASE.md:22]: I checked on their new version, and saw nothing towards that, same behavior. We can raise the issue to the project though, see what the maintainers have to say.

**Review by saitharunthodupunuru — APPROVED** [review]

**Review by miniscruff — APPROVED** [review]
